apps/system/js/telephony_settings.js

```diff
@@ -13,63 +12,46 @@
     this.connections = Array.slice(core.mobileConnections || []);
   }
 
+  TelephonySettings.SETTINGS = [
+    'ril.voicePrivacy.enabled',
+    'ril.roaming.preference',
+    'ril.clirMode',
+    'ril.radio.preferredNetworkType'
+  ];
+
   BaseModule.create(TelephonySettings, {
     name: 'TelephonySettings',
-    /**
-     * Initialzes all settings.
-     * @memberof TelephonySettings.prototype
-     */
-    _start: function() {
-      this.initVoicePrivacy();
-      this.initRoaming();
-      this.initCallerIdPreference();
-      this.initPreferredNetworkType();
-    },
 
-    /**
-     * Initializes voice privacy based on user setting.
-     */
-    initVoicePrivacy: function() {
-      var defaultVoicePrivacySettings =
-        this.connections.map(function() { return false; });
-      var voicePrivacyHelper =
-        SettingsHelper('ril.voicePrivacy.enabled', defaultVoicePrivacySettings);
-      voicePrivacyHelper.get(function got_vp(values) {
-        this.connections.forEach(function vp_iterator(conn, index) {
-          var setReq = conn.setVoicePrivacyMode(values[index]);
-          setReq.onerror = function set_vpm_error() {
-            if (setReq.error.name === 'RequestNotSupported' ||
-                setReq.error.name === 'GenericFailure') {
-              console.log('Request not supported.');
-            } else {
-              console.error('Error setting voice privacy.');
-            }
-          };
-        });
-      }.bind(this));
+    '_observe_ril.voicePrivacy.enabled' : function(values) {
+      values = values || this.connections.map(function() { return false; });
```

Comment:
The function should only be called once in the initialization instead of each time the setting changes, so are the functions for roaming, caller id, and preferred network type.